stm/stm.py

```diff
@@ -0,0 +1,418 @@
+            )
+        ds = xr.map_blocks(
+            _enrich_from_polygon_block,
+            ds,
+            args=(polygon, fields, xlabel, ylabel, type_polygon),
+            template=ds,
+        )
+
+        return ds
+
+    def _in_polygon(self, polygon, xlabel="lon", ylabel="lat"):
+        """
+        Test if points of a STM is inside a given (multi-polygon) and return result as a boolean Dask array.
+
+        Parameters
+        ----------
+        polygon : geopandas.GeoDataFrame, str, or pathlib.Path
+            Polygon or multi-polygon for query
+        xlabel : str, optional
+            Name of the x-coordinates of the STM, by default "lon"
+        ylabel : str, optional
+            Name of the y-coordinates of the STM, by default "lat"
+
+        Returns
+        -------
+        Dask.array
+            A boolean Dask array. True where points are inside the (multi-)polygon.
+        """
+
+        # Check if coords exists
+        _ = _validate_coords(self._obj, xlabel, ylabel)
+
+        # Get polygon type and the first row
+        if isinstance(polygon, gpd.GeoDataFrame):
+            type_polygon = "GeoDataFrame"
+        elif isinstance(polygon, Path) or isinstance(polygon, str):
+            type_polygon = "File"
+        else:
+            raise NotImplementedError("Cannot recognize the input polygon.")
+
+        # Enrich all fields
+        ds = self._obj
+        chunks = (ds.chunksizes["points"][0],)  # Assign an empty fields to ds
+        ds = ds.assign(
+            {
+                "mask": (
+                    ["points"],
+                    da.from_array(np.full(ds.points.shape, False), chunks=chunks),
+                )
+            }
+        )
+        mask = ds["mask"]
+        mask = xr.map_blocks(
+            _in_polygon_block,
+            mask,
+            args=(polygon, xlabel, ylabel, type_polygon),
+            template=mask,
+        )
+
+        return mask
+
+
+def _in_polygon_block(mask, polygon, xlabel, ylabel, type_polygon):
+    """
+    Block-wise function for "_in_polygon".
+    """
+    match_list, _ = _ml_str_query(mask[xlabel], mask[ylabel], polygon, type_polygon)
+    intmid = np.unique(match_list[:, 1])  # incase overlapping polygons
+    mask.data[intmid] = True
+
+    return mask
+
+
+def _enrich_from_polygon_block(ds, polygon, fields, xlabel, ylabel, type_polygon):
+    """
+    Block-wise function for "enrich_from_polygon".
+    """
+
+    # Get the match list
+    match_list, polygon = _ml_str_query(ds[xlabel], ds[ylabel], polygon, type_polygon)
+
+    _ds = ds.copy(deep=True)
+
+    if match_list.ndim == 2:
+        intuids = np.unique(match_list[:, 0])
+        for intuid in intuids:
+            intm = np.where(match_list[:, 0] == intuid)[0]
+            intmid = match_list[intm, 1]
+            for field in fields:
+                _ds[field].data[intmid] = polygon.iloc[intuid][field]
+
+    return _ds
+
+
+def _ml_str_query(xx, yy, polygon, type_polygon):
```

Comment:
this is currently based on the "contains" predicate but others are possible in principle. Maybe consider exposing the chosen predicate here via keyword, for possible use in other fct.